src/graph/optimizer/OptimizerUtils.cpp

```diff
@@ -748,34 +240,33 @@ bool getIndexColumnHintInExpr(const ColumnDef& field,
     *hint = hints.front();
     return true;
   }
-  Value begin, end;
-  if (!mergeRangeColumnHints(field, hints, &begin, &end)) {
+  std::pair<Value, bool> begin, end;
+  if (!mergeRangeColumnHints(hints, &begin, &end)) {
     return false;
   }
   ScoredColumnHint h;
   h.hint.set_column_name(field.get_name());
-  // Change scan type to prefix if begin + 1 == end
-  Value newBegin, newEnd;
-  auto status = OptimizerUtils::boundValue(ExprKind::kRelGT, begin, field, newBegin, newEnd);
-  if (!status.ok()) {
-    // TODO(yee): differentiate between empty set and invalid index to use
-    return false;
-  }
-  if (newBegin < end) {
-    // end > newBegin > begin
+  if (begin.first < end.first) {
     h.hint.set_scan_type(storage::cpp2::ScanType::RANGE);
-    h.hint.set_begin_value(std::move(begin));
-    h.hint.set_end_value(std::move(end));
+    h.hint.set_begin_value(std::move(begin.first));
+    h.hint.set_end_value(std::move(end.first));
+    h.hint.set_include_begin(begin.second);
+    h.hint.set_include_end(end.second);
     h.score = IndexScore::kRange;
-  } else if (newBegin == end) {
-    // end == neBegin == begin + 1
-    h.hint.set_scan_type(storage::cpp2::ScanType::PREFIX);
-    h.hint.set_begin_value(std::move(begin));
-    h.score = IndexScore::kPrefix;
+  } else if (begin.first == end.first) {
+    if (begin.second == false && end.second == false) {
+      return false;
+    } else {
+      h.hint.set_scan_type(storage::cpp2::ScanType::RANGE);
```

Comment:
All PREFIX has convert to [value, value]?

Reply:
No. This is only used to calculate whether the union of multiple IndexItems is an empty set (begin>end)